### ISSUE TITLE
Use publicly visible CI runs

### ISSUE THREAD

**Issue by aaronmondal** [NVIDIA/nvbench#110]
This would be a nice feature to have.

Realistically, contributors will all use different OS and/or compiler configurations and versions. So what works for one contributor won't necessarily work for another.

With public CI runs contributors would get immediate feedback on breaking changes. This way PRs can be revised to fix those breaks without a reviewer having to be present.

Since nvbench requires GPUs it may not be possible to use public pipelines for all CI builds just like that. However, a nice-to-have pipeline could be that e.g. `clang-format` runners run on all PRs via pre-commit hooks.

If it is OK for you I would like to contribute a simple proof of concept pipeline that just runs `clang-format` as a pre-merge check to verify whether PRs are conform with the `.clang-format` file in this repo.

**Comment by jrhemstad** [comment]
This is something we are actively working on. It'll be a few months yet (waiting on hardware), but it is definitely coming.

**Comment by alliepiper** [comment]
We added this a while back but missed this issue. Closing.
